### DISCUSSION TITLE
Related Studies in the VISSOFT Conference

### DISCUSSION THREAD

**Discussion by mircealungu** [git-truck/git-truck#450] [Research]
Folks,

One extra task for you in the upcoming weeks: 

Below are links to pages with all the papers published in the International Conference on Software Visualization in the last few years

- https://vissoft.info/2021/program.html 
- https://vissoft20.dcc.uchile.cl/program.html
- https://vissoft19.dcc.uchile.cl/program.html
- http://vissoft18.etsii.urjc.es/program.html
- https://vissoft17.dcc.uchile.cl/program.html
- http://vissoft16.ysu.edu/program.html
- https://vissoft.dcc.uchile.cl/program.html
- http://vissoft.iro.umontreal.ca/program.html
- https://icsm2013.tue.nl/VISSOFT/program/

I propose that you go through them and find which of them are related to your work in one of the following two ways:
  1. they are about visualizing version history 
  2. they present a study in which developers are evaluating an evolution-visualization tool

By "go through them" i don't mean read all of them. I mean, look at the title, if you can't exclude look at the abstract, if you still can't exclude, browse through the content. If it is relevant, then add an entry in this thread with: 
- title and authors
- link to paper
- brief one or two sentence summary of the paper and why is it related
- any possibilities of finding lessons for us in the paper (e.g. if it's a study with developers, did we learn something that we might take in our study?)

Does this make sense? 

M.

For most of the conference editions the papers are linked. For those where the paper is not available simply google it by the title (and authors): there is always an open version of the paper online.

**Comment by tjomson** [comment]
Just a quick look, will look at more papers later
- Scenario-Based Elicitation, Specification, and Comprehension of Transient Software Behavior (Sebastian Frank, Alireza Hakamian, Samuel Beck, Christoph Zorn André van Hoorn)
  - http://ceur-ws.org/Vol-3043/poster2.pdf
  - Seems quite different, but they are working with something about "resilience requirements", sort of like truck factor
- CodeCity: On-Screen or in Virtual Reality? (David Moreno-Lumbreras, Roberto Minelli, Andrea Villaverde, Jesus M. Gonzalez-Barahona, Michele Lanza)
  - https://www.inf.usi.ch/faculty/lanza/Downloads/More2021a.pdf
  - They test a VR code visualization tool on people from academia and inustry
- Visually Analyzing the Code Quality of Component-based Web Applications (Hagen Tarner, Daniel van den Bongard and Fabian Beck)
  - https://www.vis.wiwi.uni-due.de/fileadmin/migratedchairt3assets/file/index.pdf
  - Tool similar to Bratus. They did a user test where they let people visualize projects for the first time, to see if they can understand it, and they let the user visualize one of their own projects to see if they learn something new about their project
- Visualization of Evolution of Component-Based Software Architectures in Virtual Reality (Elke Franziska Heidmann, Lynn von Kurnatowski, Annika Meinecke and Andreas Schreiber)
  - https://conferences.computer.org/icsme/pdfs/VISSOFT2020-21GoJXGHj7IxcVsXivi2jW/991400a012/991400a012.pdf
  - Another VR tool. They did both a focus group interview, and a user study.

In general there are A LOT of papers about visualizing code as "cities" in VR

**Comment by emiljapelt** [comment]
I have look between 2013 and 2015 (inclusive).

- An Empirical Study Assessing the Effect of SeeIT 3D on Comprehension
     - https://www.researchgate.net/profile/Jairo-Aponte/publication/261072557_An_empirical_study_assessing_the_effect_of_SeeIT_3D_on_comprehension/links/55ba295d08ae092e965d9f94/An-empirical-study-assessing-the-effect-of-SeeIT-3D-on-comprehension.pdf?origin=publication_detail
    - Assesment of the usefulness of a 3D vis. in different categories, using questionaires and eyetracking.
    - Has some nice and concrete experiments. They were making test participant solve well defined goals in an unknown code base. Maybe we can use our small-scale user test to inform such goals.

- Hierarchical Software Landscape Visualization for System Comprehension: A Controlled Experiment
    - https://www.researchgate.net/profile/Wilhelm-Hasselbring/publication/301454115_Hierarchical_software_landscape_visualization_for_system_comprehension_A_controlled_experiment/links/57176aba08aeb56278c45e2a/Hierarchical-software-landscape-visualization-for-system-comprehension-A-controlled-experiment.pdf?origin=publication_detail
    - Software/System landscape vis. for comprehension
    - This is more related to software infrastructure comprehension, but the method of user test in similar to 'An Empirical Study Assessing the Effect of SeeIT 3D on Comprehension', and might give inspiration to a test design.

**Comment by emiljapelt** [comment]
I look at even more, and I believe that all years have now been looked at.

- On Using Tree Visualisation Techniques to Support Source Code Comprehension
    - https://arrow.tudublin.ie/cgi/viewcontent.cgi?article=1231&context=scschcomcon
    -  A study of using tree vis. for structural code comprehension.
    - This is what git-truck does. I will read this in full, in the coming days
